aws/resource_aws_ram_resource_share_accepter.go

```diff
@@ -0,0 +1,163 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ram"
+
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+)
+
+func resourceAwsRamResourceShareAccepter() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsRamResourceShareAccepterCreate,
+		Read:   resourceAwsRamResourceShareAccepterRead,
+		Delete: resourceAwsRamResourceShareAccepterDelete,
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(5 * time.Minute),
+			Delete: schema.DefaultTimeout(5 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"arn": {
```

Comment:
Probably just me being dumb, but where would this RAM resource share invitation ARN come from in the case where the RAM resource share is managed via Terraform? The [`AssociateResourceShare`](https://docs.aws.amazon.com/ram/latest/APIReference/API_AssociateResourceShare.html) API used by `aws_ram_principal_association` and `aws_ram_resource_association` doesn't seem to return the invitation ARN.

Reply:
Dang, that's kind of messed up. The API that generates the invite does not return the invite ARN? Playing around with it a little, it looks like the invite ARN is available by querying `GetResourceShareInvitations` in the _target_ account.

Does this accepter resource (or perhaps a new data source) then need to use the `GetResourceShareInvitations` API to get the list of invitations, and filter it by `resourceShareArn` to select `resourceShareInvitationArn`? Seems maybe prone to race conditions, between when the association is created and when the invitation can be queried from the target account.

Reply:
Since this accepter resource runs in the context of the target account then specifying `resource_share_arn` as a `Required: true` attribute should mean that [`GetResourceShareInvitations`](https://docs.aws.amazon.com/ram/latest/APIReference/API_GetResourceShareInvitations.html) would pick the one and only (hopefully 😄) `resource_shared_invitation_arn` that needs to be [accepted](https://docs.aws.amazon.com/ram/latest/APIReference/API_AcceptResourceShareInvitation.html) - both API calls would happen in the resource's _Create_ function.
Some investigation needs to be done to determine the `id` value for this resource - I think  `resource_shared_invitation_arn` would work.
You'll have to think about what it means to _Read_ and _Delete_ (plus maybe _Update_) this resource.

Reply:
> GetResourceShareInvitations would pick the one and only (hopefully 😄) resource_shared_invitation_arn that needs to be accepted

Not quite that easy... rejected invites are also returned... 😢 

```
$ aws ram get-resource-share-invitations --profile target --resource-share-arns arn:aws:ram:us-east-1:MASTER:resource-share/SHARE_UUID
{
    "resourceShareInvitations": [
        {
            "resourceShareInvitationArn": "arn:aws:ram:us-east-1:MASTER:resource-share-invitation/INVITE_UUID",
            "resourceShareName": "SHARE_NAME",
            "resourceShareArn": "arn:aws:ram:us-east-1:MASTER:resource-share/SHARE_UUID",
            "senderAccountId": "MASTER",
            "receiverAccountId": "TARGET",
            "invitationTimestamp": 1557673462.185,
            "status": "REJECTED"
        },
        {
            "resourceShareInvitationArn": "arn:aws:ram:us-east-1:MASTER:resource-share-invitation/INVITE_UUID",
            "resourceShareName": "SHARE_NAME",
            "resourceShareArn": "arn:aws:ram:us-east-1:MASTER:resource-share/SHARE_UUID",
            "senderAccountId": "MASTER",
            "receiverAccountId": "TARGET",
            "invitationTimestamp": 1557692314.324,
            "status": "PENDING"
        }
    ]
}
```

Reply:
Does seem, thankfully, as if only _one_ invite will be PENDING for a given share, though. The API accepts additional calls to associate the same share to the same principal, but there is only one PENDING invite in the target account...

Reply:
👍

Reply:
I've made it work with either invitation or share ARN.